app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -190,19 +195,30 @@
     private SessionObserver sessionObserver = new SessionObserver();
     final SessionManager.Observer managerObserver = new SessionManagerObserver(sessionObserver);
 
-    private ThemedImageButton newTabBtn;
-    private ThemedImageButton searchBtn;
-    private ThemedImageButton captureBtn;
-    private ThemedImageButton menuBtn;
     private ThemedLinearLayout toolbarRoot;
     private ThemedView bottomMenuDivider;
     private ThemedView urlBarDivider;
-    private LottieAnimationView downloadingIndicator;
-    private ImageView downloadIndicator;
     private View downloadIndicatorIntro;
+    private BottomBar bottomBar;
+    private ThemedView bottomBarBackground;
+    private ChromeViewModel chromeViewModel;
+    private BookmarkViewModel bookmarkViewModel;
+    private BottomBarViewModel bottomBarViewModel;
+    private BottomBarItemAdapter bottomBarItemAdapter;
+    private LiveData<List<BookmarkModel>> bookmarkLiveData;
 
     private long landscapeStartTime = 0L;
 
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        BookmarkViewModel.Factory factory = new BookmarkViewModel.Factory(
+                BookmarkRepository.getInstance(BookmarksDatabase.getInstance(requireContext())));
+        bookmarkViewModel = ViewModelProviders.of(this, factory).get(BookmarkViewModel.class);
+        bottomBarViewModel = Inject.obtainBottomBarViewModel(getActivity());
```

Comment:
getActivity() here might be null in fragment re-create scenario. Maybe you should consider moving this to onActivityCreated(), where getActivity() is guaranteed to be valid.